sisl/viz/plotly/input_fields/queries.py

```diff
@@ -315,6 +337,13 @@ def _split_query(self, query, on, only=None, exclude=None, query_gen=None, **kwa
             request that this method has come up with and gets a chance to do some modifications.
 
             This may be useful, for example, to give each request a color, or a custom name.
+        ignore_constraints: boolean or array-like, optional
+            determines whether constraints (existing in the query) on the parameters that we want 
+            to split along should be taken into consideration.
+
+            If `False`: all constraints considered.
+            If `True`: no constraints considered.
+            If array-like: parameters contained in the list ignore their constraints.
```

Comment:
while this seems to solve the problem, I still don't get why ;)
I need to look into your request handles. Perhaps they could be restructured for the same effect with less clutter.

In particular it seems odd that the original one added l=1 when l=0 was requested explicitily.

Reply:
What do you mean the original one? `ignore_constraints=True` is not "the original one", the default value of `ignore_constraints` is `False`.

What seems odd about ignoring a constraint if the user told us to ignore it? hahaha

Reply:
It just seems obvious that `l=0` would ignore different `l` values otherwise.

Again, you have to see me as a new user, and a new user would not see this connection :)

Reply:
Yes, that's why the **default behavior is to ignore the other `l` values**, therefore applying the constraint of `l=0` because the request that we are splitting imposes that constraint on `l`.

Only when you say `ignore_constraints=True` is it that `l` becomes free to be any value when you split on it.

I don't understand what the problem is :confused:

Reply:
The problem is that users (including me) don't understand what `ignore_constraints` may be used for?

And to me it seems that ignore_constraints should always be `False`, when is `True` useful?

Also, from the doc's in `split_query` there is no other mention of `constraints`. So it is not clear what it refers to? Is it used together with `on` or something else? And also, how is it connected to `kwargs`.
:)

Reply:
> when is True useful

Mainly in non-colinear spin PDOS, that's why I introduced it. Automatically you get a request that has `spin=["sum"]` unless you specify a channel, because that's what you want (you don't want "sum" + "x" + "y", + "z"). But then if you split that on `spin`, you have the constraint that spin can only be `"sum"`, so basically you wouldn't split anything. That's why you need to ignore that constraint if you want the `x`, `y` and `z` channels for that request.

It may also be useful when you have a request like `{"atoms": [0, 1, 5, 8], "n": [3], "l": [0], "zeta": [1]}` and you want to see the DOS of that same request across `l`, for example. 

But the main point is to overcome the fact that splitting `spin=["sum"]` doesn't result in `spin=["x", "y", "z"]`.

> Also, from the doc's in split_query there is no other mention of constraints. So it is not clear what it refers to? Is it used together with on or something else? And also, how is it connected to kwargs.
> :)

In the `PdosPlot.split_requests` method:

> ignore_constraints: boolean or array-like, optional
>     
> determines whether constraints (**existing in the request to be splitted**) 
> on the parameters that we want to split along should be taken into consideration.
> If `False`: all constraints considered.
> If `True`: no constraints considered.
> If array-like: parameters contained in the list ignore their constraints.

How would you say it? Maybe "imposed by the request to be splitted"?

I don't know, this is a complex use case, and you will only come across it if you need to use it, where you will already understand what your problem is. Otherwise you don't even realize it's there.

Reply:
It seems that I am missing the implementation details. I still have no idea what you are doing ;)

I'll have to read in the queries file regarding these details, I guess?

From the looks of it, it is very complicated :)
Why wouldn't users just add `spin=["sum", "x", ...]` explicitly when they want so?

Reply:
Yes I agree that the `OrbitalQueries` input is not simple, but that's the way I managed to handle all the corner cases and make queries behave like the user would expect. (note that is not used only by `PdosPlot`, also by `FatbandsPlot` and I'm quite happy with the consistency it achieved).

Maybe you find an easier way of handling all this :)

Reply:
Ok. There is just lots to it :) And new for me, so lots of questions. It is very difficult to generalize and make things simple in a GUI interface like this :)
So all good :)